libs/common/src/lib/i18n/i18n.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core'
+import { TranslateLoader } from '@ngx-translate/core'
+import { forkJoin, Observable } from 'rxjs'
+import { HttpClient } from '@angular/common/http'
+import { StandardsApiService } from '@lib/gn-api'
+import { map } from 'rxjs/operators'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TranslationService implements TranslateLoader {
+  constructor(
+    private http: HttpClient,
+    private standardApiService: StandardsApiService
+  ) {}
+
+  getTranslation(lang: string): Observable<any> {
+    const observables: Observable<any>[] = []
+    const translationFiles = [`assets/i18n/${lang}.json`]
+
+    const core = this.http.get(translationFiles[0])
+    observables.push(core)
+    observables.push(
```

Comment:
How does it work exactly ? does it fetch another translation file ?
Why do we still have `facets.block.title.cl_hierarchyLevel.key` in the assets ?

Reply:
See https://apps.titellus.net/geonetwork/srv/api/standards/iso19139/codelists?codelist=gmd%3AMD_TopicCategoryCode&codelist=gmd%3AMD_ScopeCode&codelist=gmd%3AMD_MaintenanceFrequencyCode&codelist=gmd%3AMD_ProgressCode&codelist=gmd%3ADS_InitiativeTypeCode&codelist=gmd%3AMD_SpatialRepresentationTypeCode

The API call loads the codelist values translations. Not its label. If we want to load codelist labels in one call, we may need to extend the API...